### PR TITLE
Rewrite line renderer: Part 2 (remove LineRenderer plus other cleanups)

### PR DESCRIPTION
**Pull request by pchote**
Followup to and depends on #10148 (which appears as the first 10 commits in this pr).

The main visual change here is the ability to draw connected line strips, which fixes the holes that can appear at the corners of rectangles and the beacon triangle.  It also fixes the position of the voxel shadow bounding box in the rendering geometry visualization.

### CODE REVIEW

File: OpenRA.Game/Graphics/RgbaColorRenderer.cs [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+
+			vertices[nv++] = new Vertex(start - corner + Offset, r, g, b, a);
+			vertices[nv++] = new Vertex(start + corner + Offset, r, g, b, a);
+			vertices[nv++] = new Vertex(end + corner + Offset, r, g, b, a);
+			vertices[nv++] = new Vertex(end - corner + Offset, r, g, b, a);
+		}
+
+		/// <summary>
+		/// Calculate the intersection of two lines.
+		/// Will behave badly if the lines are parallel
```
Comment (Phrohdoh) [inline OpenRA.Game/Graphics/RgbaColorRenderer.cs]: "badly" as in what?

File: OpenRA.Game/Graphics/RgbaColorRenderer.cs [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+			vertices[nv++] = new Vertex(c + Offset, cr, cg, cb, ca);
+			vertices[nv++] = new Vertex(d + Offset, cr, cg, cb, ca);
+		}
+
+		public void FillEllipse(RectangleF r, Color color, int vertices = 32)
+		{
+			// TODO: Create an ellipse polygon instead
+			var a = (r.Right - r.Left) / 2;
+			var b = (r.Bottom - r.Top) / 2;
+			var xc = (r.Right + r.Left) / 2;
```
Comment (Phrohdoh) [inline OpenRA.Game/Graphics/RgbaColorRenderer.cs]: What do the `c` suffixes indicate here?
If that is "change" can then become `d` for delta (consistency)?

File: OpenRA.Game/Graphics/RgbaColorRenderer.cs [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+			{
+				var dx = a * (float)Math.Sqrt(1 - (y - yc) * (y - yc) / b / b);
+				DrawLine(new float2(xc - dx, y), new float2(xc + dx, y), 1, color);
+			}
+		}
+
+		public void SetViewportParams(Size screen, float zoom, int2 scroll)
+		{
+			shader.SetVec("Scroll", scroll.X, scroll.Y);
+			shader.SetVec("r1", zoom * 2f / screen.Width, -zoom * 2f / screen.Height);
```
Comment (Phrohdoh) [inline OpenRA.Game/Graphics/RgbaColorRenderer.cs]: Can `r1` and `r2` be a bit more verbose?

File: OpenRA.Game/Graphics/RgbaColorRenderer.cs (hunk 1 above)
Comment (pchote) [inline OpenRA.Game/Graphics/RgbaColorRenderer.cs]: If `da ≈ db` then `d` tends to 0 and the intersection blows up.

**Comment by pchote** [comment]
Rebased on top of now-merged dependencies.

**Comment by pchote** [comment]
Rebased.

**Comment by Phrohdoh** [comment]
This all looks sensible to me, but I'd like someone with a better mathematics background to review those parts.
I'll give this a test ingame now.

**Comment by Mailaender** [comment]
Can't spot regressions in rain effects, contrails, graphs and target lines. Tested possible clashes with zooming as well. :white_check_mark: You gave the more difficult parts of our code base some badly needed explanatory comments. :+1:

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/7d67c6a64c87a24f15fb5ce9a07a5a50970f19e3...ec7ecf178936d4b22e4f4bba0f2478b9924369c0)

**Pull request merged**